src/kagglehub/auth.py

```diff
@@ -9,20 +12,96 @@
 
 INVALID_CREDENTIALS_ERROR = 403
 
+NOTEBOOK_LOGIN_TOKEN_HTML_START = """<center> <img
+src=https://www.kaggle.com/static/images/site-logo.png
+alt='Kaggle'> <br> Copy a token from <a
+href="https://www.kaggle.com/settings/account" target="_blank">your Kaggle
+settings page</a> and paste it below. <br> </center>"""
 
-def login(validate_credentials=True):  # noqa: FBT002
-    """Prompt the user for their Kaggle username and API key and save them globally."""
+NOTEBOOK_LOGIN_TOKEN_HTML_END = """
+<b>Thank You</b></center>"""
 
-    username = input("Enter your Kaggle username: ")
-    api_key = input("Enter your Kaggle API key: ")
 
-    set_kaggle_credentials(username=username, api_key=api_key)
+@contextmanager
+def capture_logger_output():
+    """Capture output that is logged using the logger.
+
+    Example:
+    ```py
+    >>> output = capture_logger_output()
+    >>> logger.info("This is an info message")
+    >>> logger.error("This is an error message")
+    >>> print(output)
+    ```
+    """
+    buffer = io.StringIO()
+    handler = logging.StreamHandler(buffer)
+    logger = logging.getLogger()
+    logger.addHandler(handler)
+    try:
+        yield buffer
+    finally:
+        logger.removeHandler(handler)
 
-    logger.info("Kaggle credentials set.")
 
-    if not validate_credentials:
-        return
+def is_in_notebook():
+    return "IPython" in sys.modules
+
+
+def notebook_login(validate_credentials) -> None:
+    """Prompt the user for their Kaggle token and save it in a widget (Jupyter or Colab)."""
+    library_error = "You need the `ipywidgets` module: `pip install ipywidgets`."
+    try:
+        from IPython.display import display  # type: ignore
+        from ipywidgets import widgets  # type: ignore
+    except ImportError:
+        raise ImportError(library_error)  # noqa: B904
+
+    box_layout = widgets.Layout(display="flex", flex_flow="column", align_items="center", width="50%")
+
+    username_widget = widgets.Text(description="Username:")
+    token_widget = widgets.Password(description="Token:")
+    login_button = widgets.Button(description="Login")
+
+    login_token_widget = widgets.VBox(
+        [
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_START),
+            username_widget,
+            token_widget,
+            login_button,
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_END),
+        ],
+        layout=box_layout,
+    )
+    display(login_token_widget)
+
+    # On click events
```

Comment:
Instead of having a comment here, rename the method to be more clear. Maybe:

`def on_click_login_button(t):`